Sources/MongoSwift/ConnectionString.swift

```diff
@@ -93,14 +93,14 @@ internal class ConnectionString {
         }
 
         // libmongoc does not validate negative timeouts in the input string, so do it ourselves here.
-        if let connectTimeoutMS = self.options?[MONGOC_URI_CONNECTTIMEOUTMS]?.int32Value, connectTimeoutMS < 1 {
+        if let connectTimeoutMS = self.options?[MONGOC_URI_CONNECTTIMEOUTMS]?.int32Value, connectTimeoutMS < 0 {
```

Comment:
oh no.... I'm sorry to go back and forth on this, but your change here made me look into why I'd done this in the first place.

 libmongoc does not comply with the spec: it does _not_ treat a value of 0 for any timeout option as "no timeout", but instead treats it as "use the default".

see [here](http://mongoc.org/libmongoc/current/mongoc_uri_t.html#connection-options):
> Setting any of the *TimeoutMS options above to 0 will be interpreted as “use the default value”.

since its behavior when 0 is specified is non-spec compliant, I think it would be better for us to not allow a value of `0` for these options after all, and go back to your original code.

in 2.0 we can change our behavior to be in line with the spec.

I should have opened a ticket about this but I don't think I did. Could you please open a ticket along the lines of "allow users to specify 0 for timeout values"?

Reply:
I think this still needs to be changed back.

Reply:
changed back